misk-inject/src/test/kotlin/misk/inject/InstallOnceTest.kt

```diff
@@ -0,0 +1,40 @@
+package misk.inject
+
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@MiskTest
+class InstallOnceTest {
+  @MiskTestModule
+  val module = TestInstallModule()
+
+  @Inject @TestAnnotation private lateinit var installedOnce: String
+
+  @Test
+  fun testInstallOnce() {
+    Assertions.assertThat(installedOnce).startsWith("only one of me with identity")
```

Comment:
Doesn't this test need to also print the identity result? (i.e. `${System.identityHashCode(this)}`)

Reply:
startsWith?

I just have to put something that is different with each instance. Because otherwise String already overloads equals and hashcode to be the same with the same contents.